scheduler/spread.go

```diff
@@ -193,20 +182,31 @@ func evenSpreadScoreBoost(pset *propertySet, option *structs.Node) float64 {
 			maxCount = value
 		}
 	}
+
+	// calculate boost based on delta between the current and the minimum
+	deltaBoost := 1.0
+	if minCount == 0 {
+		deltaBoost = -1.0
+	} else {
+		delta := int(minCount - currentAttributeCount)
+		deltaBoost = float64(delta) / float64(minCount)
+	}
 	if currentAttributeCount < minCount {
-		// Small positive boost for attributes with min count
-		return evenSpreadBoost
+		// positive boost for attributes with min count
```

Comment:
The first two cases can be simplified to `if currentAttributeCount != minCount {`